core/src/main/java/org/jclouds/http/config/SSLModule.java

```diff
@@ -43,8 +44,8 @@
    @Override
    protected void configure() {
       bind(HostnameVerifier.class).annotatedWith(Names.named("untrusted")).to(LogToMapHostnameVerifier.class);
-      bind(new TypeLiteral<Supplier<SSLContext>>() {
-      }).annotatedWith(Names.named("untrusted")).to(new TypeLiteral<UntrustedSSLContextSupplier>() {
+      bind(new TypeLiteral<Supplier<SSLSocketFactory>>() {
+      }).annotatedWith(Names.named("untrusted")).to(new TypeLiteral<UntrustedSSLSocketFactorySupplier>() {
```

Comment:
I'm a bit concerned about this change. There are still uses of the SSLContext injection. For example, take a look at the AapcheHC or the OkHttp drivers. Those would need to be updated too.

I can also imagine that there can also be users that are providing a custom version of the SSLContext, to customize how it behaves, and I think we should try not to break them. Is there any way you can introduce your changes by keeping the old supplier? Would it be possible to have both, the old one and your SSLSocketFactory new one, and let your supplier use the injected SSLContext one? This way we keep backward-compatibility and the only thing we care about is our codebase using the factory supplier instead of directly getting it from the context.

Reply:
I understand your concern. I could have JavaUrlHttpCommandExecutorService accept both a nullable Supplier<SSLContext> and a Supplier<SSLSocketFactory> for the untrusted use case as a fallback - this will be in line with the existing behavior and shouldn't break existing users providing SSLContext. In this case though, I suspect that only the untrusted use case will be using the socket cache.

I'm not sure if the untrusted SSLSocketFactory supplier should build upon another injected SSLContext supplier since these are currently mutually exclusive use cases in JavaUrlHttpCommandExecutorService (and also I don't think SSLContext can be init-ed twice).